### PR TITLE
Split section wallet

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

**Review by madadam — DISMISSED** [review]
Looks good, just delete that commented out old pieces of old code.

**Review by madadam — APPROVED** [review]

File: src/routing/approved.rs [hunk 1]
```diff
@@ -1778,8 +1788,20 @@ impl Approved {
                 commands.extend(self.send_sync(self.section.clone(), self.network.clone())?);
             }
 
-            let sibling_key = if new_prefix != old_prefix {
-                self.section_key(&new_prefix.sibling()).copied()
+            let sibling_elders = if new_prefix != old_prefix {
+                if let Some(sibling_key) = self.section_key(&new_prefix.sibling()).copied() {
+                    if let Some(info) = self.network.get(&new_prefix.sibling()).cloned() {
```
Comment (maqi) [inline src/routing/approved.rs:1793]: no need to use `cloned` ? 
you can use `let Some(ref info) = self.network.get(&new_prefix.sibling())` ?

same to the above `Some(sibling_key)` . only clone when assigning?

File: src/routing/approved.rs [hunk 2]
```diff
@@ -2135,6 +2161,7 @@ impl Approved {
         };
         let mut commands = vec![];
 
+        // consider removing this, we are getting duplciate msgs by it
```
Comment (maqi) [inline src/routing/approved.rs:2164]: put a `TODO:` keyword

File: examples/minimal.rs [hunk 3]
```diff
@@ -223,6 +223,9 @@ fn run_node(index: usize, mut node: Routing, mut event_stream: EventStream) -> J
 // Handles the event emitted by the node.
 async fn handle_event(index: usize, node: &mut Routing, event: Event) -> bool {
     match event {
+        Event::Genesis => {
+            info!("Node #{} is Genesis", index);
```
[the remark this answers is not in the document]
Comment (joshuef) [reply examples/minimal.rs:227]: I believe so yep.

**Review by maqi — APPROVED** [review]

**Pull request merged**